python/lsst/ctrl/iip/Credentials.py

```diff
@@ -0,0 +1,88 @@
+# This file is part of ctrl_iip
+# 
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import os
+import os.path
+import stat
+import sys
+import yaml
+from lsst.ctrl.iip.ThreadManager import ThreadManager
+
+class Credentials:
+
+    def __init__(self, filename):
+        self.cred = self.loadSecureFile(filename)
+    
+    def getUser(self, user_alias):
+        return self.cred['rabbitmq_users'][user_alias]
+
+    def getPasswd(self, passwd_alias):
+        return self.cred['rabbitmq_users'][passwd_alias]
+
+    def loadSecureFile(self, cred_file):
+        """load a secure YAML file"""
+
+        home = os.environ["HOME"]
+        config_dir = os.path.join(home,".lsst")
+        # check for the existence of the configuration directory
+        if os.path.isdir(config_dir):
+            stat_info = os.stat(config_dir)
+            mode = stat_info.st_mode
+            
+            # check that the permissions are set to rwx for only the user
+            if (mode & (stat.S_IWOTH | stat.S_IWGRP | stat.S_IROTH | stat.S_IRGRP)):
+                print("directory '%s' is unsecure.  Run 'chmod 700 %s' to fix this." % (config_dir, config_dir))
```

Comment:
In order to do that, I'll have to rearrange some things because Logging isn't set up at that point. The credentials currently load first, then the Configuration file, and the Logging gets set up based on things in the Configuration file.  At the time I wrote this, I thought checking the iip_cred.yaml file was important to get out of the way first, but I'm not sure it makes that big of a difference.  I'll look into rearranging things.